File: two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = {}
        
        for i in range(len(nums)):
            val = target - nums[i]
            if val in hashmap:
                return i, hashmap[val]
            hashmap[nums[i]] = i

File: twoSum.js
var twoSum = function(nums, target){
    const map = new Map()
    for(let i = 0; i < nums.length; i++){
        const num = nums[i]
        const val = target - num
        const sumIndex = map.get(val)

        const isTarget = map.has(val)
        if(isTarget){
            return [i, sumIndex]
        }
        map.set(num, i)
    }
}